synapse/storage/schema/main/delta/75/01_per_user_experimental_features.sql

```diff
@@ -0,0 +1,34 @@
+/* Copyright 2023 The Matrix.org Foundation C.I.C
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+-- Table containing a list of experimental features and whether they are
+-- enabled for a given user
+CREATE TABLE per_user_experimental_features (
+    -- The User ID to check/set the feature for
+    user_id TEXT NOT NULL PRIMARY KEY,
```

Comment:
Should we make this a foreign key?

Reply:
So I don't know much about foreign keys other than it has to do with data integrity across tables - what would be the reasoning/benefit to making this a foreign key (I would imagine that would connect it to the `users` table somehow?)

Reply:
yeah, the text you want to add here is `REFERENCES users (name)` after `PRIMARY KEY`.

What it means is that you can't insert an experimental feature flag for a user that doesn't exist in the users table.